### PR TITLE
feat: binary decoupling between OCC and RefApp

### PR DESCRIPTION
**Pull request by OlyaAn**
[AB#65200](https://dev.azure.com/orckestra001/da965986-c85b-43be-9915-219568b000e5/_workitems/edit/65200)

### CODE REVIEW

File: src/Orckestra.Composer.MyAccount/Repositories/CustomerAddressRepository.cs [hunk 1]
```diff
@@ -44,14 +43,40 @@ public virtual Task<List<Address>> GetCustomerAddressesAsync(Guid customerId, st
...
+            request.Email = address.Email;
+            request.Notes = address.Notes;
+            request.ScopeId = scope;
+            request.IsPreferredBilling = address.IsPreferredBilling;
+            request.IsPreferredShipping = address.IsPreferredShipping;
+
+            if (address.PropertyBag != null)
+            {
+                request.PropertyBag = new PropertyBag(address.PropertyBag);
+            }
```
Comment (InnaBoitsun) [inline src/Orckestra.Composer.MyAccount/Repositories/CustomerAddressRepository.cs:78]: some code is duplicated - create some extention for this

File: src/Orckestra.Composer.OccClient/Orckestra.Composer.OccClient.csproj [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  <PropertyGroup>
+    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
+    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
+    <ProjectGuid>{06196B4D-24A8-4BDE-B093-E56B9638DE66}</ProjectGuid>
+    <OutputType>Library</OutputType>
+    <AppDesignerFolder>Properties</AppDesignerFolder>
+    <RootNamespace>Orckestra.Composer.OccClient</RootNamespace>
+    <AssemblyName>Orckestra.Composer.OccClient</AssemblyName>
+    <TargetFrameworkVersion>v4.7.2</TargetFrameworkVersion>
+    <FileAlignment>512</FileAlignment>
```
Comment (InnaBoitsun) [inline src/Orckestra.Composer.OccClient/Orckestra.Composer.OccClient.csproj:13]: solution projects have 4.8 framework, use the same

File: src/Orckestra.Composer.OccClient/Properties/PublishProfiles/Package.pubxml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    <WebPublishMethod>FileSystem</WebPublishMethod>
+    <LastUsedBuildConfiguration>Release</LastUsedBuildConfiguration>
+    <LastUsedPlatform>Any CPU</LastUsedPlatform>
+    <SiteUrlToLaunchAfterPublish />
+    <LaunchSiteAfterPublish>True</LaunchSiteAfterPublish>
+    <ExcludeApp_Data>False</ExcludeApp_Data>
+    <publishUrl>.\_package</publishUrl>
+    <DeleteExistingFiles>True</DeleteExistingFiles>  
+  </PropertyGroup>
+</Project>
```
Comment (InnaBoitsun) [inline src/Orckestra.Composer.OccClient/Properties/PublishProfiles/Package.pubxml:17]: is it needed?

**Review by InnaBoitsun — CHANGES_REQUESTED** [review]

File: src/Orckestra.Composer/Caching/ICacheClientV2.cs [hunk 4]
```diff
@@ -0,0 +1,9 @@
+﻿using System;
+
+namespace Orckestra.Composer.Caching
+{
+    public interface ICacheClientV2 : ICacheClient
+    {
```
[the remark this answers is not in the document]
Comment (OlyaAn) [reply src/Orckestra.Composer/Caching/ICacheClientV2.cs:6]: yes, in the CacheProvider.cs

**Review by InnaBoitsun — APPROVED** [review]

**Pull request merged**